### DISCUSSION TITLE
ReferenceError: document is not defined

### DISCUSSION THREAD

**Discussion by brillout** [vikejs/vike#766] [Discussion]
Moved from Discord:

> Hi when using the plugin I get this error:
ReferenceError: document is not defined
I read the log and everything seems to be because of react-router, can someone help me?

**Comment by brillout** [comment]
This doesn't seem to be related to VPS?

**Comment by brillout** [comment]
> Can you explain to me the meaning of VPS?

VPS = vite-plugin-ssr

**Comment by plancerio** [comment]
Ok, sorry for not being so specific, it turns out that in one of my components I was using BrowserRouter and I think it is not compatible with VPS or so I think, I removed it from the component and stopped having that problem, but two more errors appeared, the first was this:
ReferenceError: window is not defined

AND:
ReferenceError: sessionStorage is not defined

Does it have something to do with vps or do I need to integrate something?

**Reply by brillout** [reply]
I don't think these are related to VPS either.

Make sure you don't load client-only code in the server.
